setuptools/dist.py

```diff
@@ -52,6 +51,9 @@
 if TYPE_CHECKING:
     from typing_extensions import TypeAlias
 
+    from pkg_resources import Distribution as _pkg_resources_Distribution
```

Comment:
It is good this is inside a `if TYPE_CHECKING` block. Thank you!

Ideally we want to avoid importing `pkg_resources` in runtime because of the performance hit.

It is only acceptable to "lazily" import `pkg_resources` on deprecated code paths (but never eagerly).